File: arrays/concatenationOfArray.java
package arrays;
// https://leetcode.com/problems/concatenation-of-array/ 

public class concatenationOfArray {
    public static void main(String[] args) {
        SolutionCOA solution = new SolutionCOA ();
        System.out.println(solution.getConcatenation(new int[]{1,3,2,1}));
    }
    
}

class SolutionCOA {
    public int[] getConcatenation(int[] nums) {
        int[] result = new int[nums.length*2];
        for (int i=0; i<=nums.length-1; i++){
            //result[i] = nums[i%nums.length];
            //One way to do it

            //Other way
            result[i] = result[nums.length+1] = nums[i];
        }
        return result;
        
        
    }
}

File: arrays/concatenationOfArray.py
#https://leetcode.com/problems/concatenation-of-array/

def getConcatenation(nums):
    #1: brute force
    # for i in range(len(nums)):
    #     nums.append(nums[i])
    # return nums
    
    #2: concatenate
    return nums+nums

    #3: using extend
    # nums.extend(nums)
    # return nums

    #4: list comprehension
    # return [i for x in range(2) for i in nums]
    
    #5: unpaking list items
    # return [*nums]*2

#input
nums = [1,2,1]
print(getConcatenation(nums))